src/Common/tests/Tests/System/SpanTestHelpers.cs

```diff
@@ -26,5 +28,21 @@ public static class SpanTestHelpers
                 Assert.False(true, "Wrong exception thrown: Expected " + typeof(E).GetType() + ": Actual: " + wrongException.GetType());
             }
         }
+
+        public static void AssertThrows<E, T>(Span<T> span, AssertThrowsAction<T> action) where E : Exception
+        {
+            try
+            {
+                action(span);
+                Assert.False(true, "Expected exception: " + typeof(E).GetType());
+            }
+            catch (E)
```

Comment:
Nit: this is functionally incongruent from Assert.Throws, which checks the exact type of E; this is more in line with ThrowsAny.